src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPEditorWrapper.java

```diff
@@ -0,0 +1,246 @@
+
+    private final Editor editor;
+
+    private final EditorMouseMotionListener editorMouseMotionListener;
+
+    private final CaretListener caretListener;
+
+    private final DocumentListener documentListener;
+
+    private boolean documentChanged;
+
+    private LSPCodeActionResult codeActionResult;
+
+    private LSPEditorWrapper(Editor editor) {
+        this.editor = editor;
+        this.editorMouseMotionListener = new EditorMouseMotionListener() {
+            @Override
+            public void mouseMoved(@NotNull EditorMouseEvent e) {
+                Document document = editor.getDocument();
+                VirtualFile file = LSPIJUtils.getFile(document);
+                List<Diagnostic> currentDiagnostics = LSPVirtualFileWrapper.getLSPVirtualFileWrapper(file).getAllDiagnostics();
+                if (currentDiagnostics != null && !currentDiagnostics.isEmpty()) {
+                    int offset = getTargetOffset(e);
+                    if (offset != -1) {
+                        for (Diagnostic diagnostic : currentDiagnostics) {
+                            int startOffset = LSPIJUtils.toOffset(diagnostic.getRange().getStart(), document);
+                            int endOffset = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(), document);
+                            if (offset >= startOffset && offset <= endOffset) {
+                                try {
+                                    updateQuickFixAt(offset,startOffset,endOffset).get();
+                                } catch (InterruptedException interruptedException) {
+                                    interruptedException.printStackTrace();
+                                } catch (ExecutionException executionException) {
+                                    executionException.printStackTrace();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        };
+        this.editor.addEditorMouseMotionListener(editorMouseMotionListener);
+
+        this.documentChanged = false;
+        this.documentListener = new DocumentListener() {
+
+            @Override
+            public void documentChanged(@NotNull DocumentEvent event) {
+                documentChanged = true;
+            }
+        };
+        this.editor.getDocument().addDocumentListener(documentListener);
+
+        this.caretListener = new CaretListener() {
+            @Override
+            public void caretPositionChanged(@NotNull CaretEvent event) {
+                if (documentChanged) {
+                    documentChanged = false;
+                    return;
+                }
+                int offset = editor.getCaretModel().getCurrentCaret().getOffset();
+                VirtualFile file = LSPIJUtils.getFile(editor.getDocument());
+                if (file != null) {
+                    updateQuickFixAt(offset, -1,-1);
+                }
+            }

```

Comment:
nitpicking: I believe that code readability would gain a lot if the 3 anonymous callbacks above would be separated into their own methods or inner classes